solution_template/jenkins/nested/VM-password.json

```diff
@@ -0,0 +1,147 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "outputs": {
+  },
+  "parameters": {
+    "adminPassword": {
+      "metadata": {
+        "description": "Password for the Virtual Machine. Will be used only is authenticationType is 'password'"
+      },
+      "type": "securestring"
+    },
+    "adminSSHPublicKey": {
+      "metadata": {
+        "description": "Public SSH key for the Virtual Machine. The parameter is ignored."
+      },
+      "type": "securestring"
+    },
+    "adminUserName": {
+      "metadata": {
+        "description": "User name for the Virtual Machine"
+      },
+      "type": "string"
+    },
+    "extensionCommandToExecute": {
+      "metadata": {
+        "description": "Extension execute command"
+      },
+      "type": "string"
+    },
+    "extensionFileUri": {
+      "metadata": {
+        "description": "Extension File Uri"
+      },
+      "type": "string"
+    },
+    "networkInterfaceId": {
+      "metadata": {
+        "description": "Network Interface Id"
+      },
+      "type": "string"
+    },
+    "osDiskVHDUri": {
+      "metadata": {
+        "description": "OS Disk VHD URI"
+      },
+      "type": "string"
+    },
+    "ubuntuSku": {
+      "metadata": {
+        "description": "Ubuntu SKU"
+      },
+      "type": "string"
+    },
+    "vmLocation": {
+      "metadata": {
+        "description": "Virtual Machine Location"
+      },
+      "type": "string"
+    },
+    "vmName": {
+      "metadata": {
+        "description": "Virtual Machine Name"
+      },
+      "type": "string"
+    },
+    "vmSize": {
+      "metadata": {
+        "description": "Virtual Machine Size"
+      },
+      "type": "string"
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2016-03-30",
+      "dependsOn": [],
+      "location": "[parameters('vmLocation')]",
+      "name": "[parameters('vmName')]",
+      "properties": {
+        "hardwareProfile": {
+          "vmSize": "[parameters('vmSize')]"
+        },
+        "networkProfile": {
+          "networkInterfaces": [
+            {
+              "id": "[parameters('networkInterfaceId')]"
+            }
+          ]
+        },
+        "osProfile": {
+          "adminPassword": "[parameters('adminPassword')]",
+          "adminUsername": "[parameters('adminUserName')]",
+          "computerName": "[parameters('vmName')]"
+        },
+        "storageProfile": {
+          "imageReference": {
+            "offer": "UbuntuServer",
+            "publisher": "Canonical",
+            "sku": "[parameters('ubuntuSku')]",
+            "version": "latest"
+          },
+          "osDisk": {
+            "caching": "ReadWrite",
+            "createOption": "FromImage",
+            "name": "osDisk",
+            "vhd": {
+              "uri": "[parameters('osDiskVHDUri')]"
+            }
+          }
+        }
+      },
+      "resources": [
+        {
```

Comment:
Consider moving the vmExtension resource to the main template to reduce the amount of copied code (and simplify the parameters you have to pass)